Mathlib/Order/Directed.lean

```diff
@@ -250,18 +250,66 @@ theorem isBot_iff_isMin [IsDirected α (· ≥ ·)] : IsBot a ↔ IsMin a :=
 theorem isTop_iff_isMax [IsDirected α (· ≤ ·)] : IsTop a ↔ IsMax a :=
   ⟨IsTop.isMax, IsMax.isTop⟩
 
-variable (β) [PartialOrder β]
+end Preorder
+
+-- see Note [lower instance priority]
+instance (priority := 100) SemilatticeSup.to_isDirected_le [SemilatticeSup α] :
+    IsDirected α (· ≤ ·) :=
+  ⟨fun a b => ⟨a ⊔ b, le_sup_left, le_sup_right⟩⟩
+
+-- see Note [lower instance priority]
+instance (priority := 100) SemilatticeInf.to_isDirected_ge [SemilatticeInf α] :
+    IsDirected α (· ≥ ·) :=
+  ⟨fun a b => ⟨a ⊓ b, inf_le_left, inf_le_right⟩⟩
+
+-- see Note [lower instance priority]
+instance (priority := 100) OrderTop.to_isDirected_le [LE α] [OrderTop α] : IsDirected α (· ≤ ·) :=
+  ⟨fun _ _ => ⟨⊤, le_top _, le_top _⟩⟩
+
+-- see Note [lower instance priority]
+instance (priority := 100) OrderBot.to_isDirected_ge [LE α] [OrderBot α] : IsDirected α (· ≥ ·) :=
+  ⟨fun _ _ => ⟨⊥, bot_le _, bot_le _⟩⟩
```

Comment:
Can you put this back to the end of the file? The first two are about `Semilattice...` which is stronger than `PartialOrder`, and I don't see why you need to move the last two?

Reply:
Yeah, I don't remember why I moved these. I'll move them back.